docs/logging_snippets.py

```diff
@@ -0,0 +1,413 @@
+    """List entries via a client across projects."""
+
+    # [START client_list_entries_multi_project]
+    PROJECT_IDS = ['friendly-coconut-337', 'precious-gravel-232']
+    entries, token = client.list_entries(projects=PROJECT_IDS)  # API request
+    while True:
+        for entry in entries:
+            do_something_with(entry)
+        if token is None:
+            break
+        entries, token = client.list_entries(
+            projects=PROJECT_IDS, page_token=token)  # API request
+    # [END client_list_entries_multi_project]
+
+
+@snippet
+def client_list_entries_filt(
+        client, to_delete):  # pylint: disable=unused-argument
+    """List entries via a client, filtered"""
+
+    # [START client_list_entries_filter]
+    FILTER = "logName:my-log AND textPayload:simple"
+    entries, token = client.list_entries(filter_=FILTER)  # API request
+    while True:
+        for entry in entries:
+            do_something_with(entry)
+        if token is None:
+            break
+        entries, token = client.list_entries(
+            filter_=FILTER, page_token=token)  # API request
+    # [END client_list_entries_filter]
+
+
+@snippet
+def client_list_entries_desc(
+        client, to_delete):  # pylint: disable=unused-argument
+    """List entries via a client, descending"""
+
+    # [START client_list_entries_desc]
+    from google.cloud.logging import DESCENDING
+    entries, token = client.list_entries(order_by=DESCENDING)  # API request
+    while True:
+        for entry in entries:
+            do_something_with(entry)
+        if token is None:
+            break
+        entries, token = client.list_entries(
+            order_by=DESCENDING, page_token=token)  # API request
+    # [END client_list_entries_desc]
+
+
+@snippet
+def client_list_entries_size(
+        client, to_delete):  # pylint: disable=unused-argument
+
+    """List entries via a client, w/ custom page size"""
+
+    # [START client_list_entries_page_size]
+    entries, token = client.list_entries(page_size=100)  # API request
+    while True:
+        for entry in entries:
+            do_something_with(entry)
+        if token is None:
+            break
+        entries, token = client.list_entries(
+            page_size=100, page_token=token)  # API request
+    # [END client_list_entries_page_size]
+
+
+@snippet
+def metric_crud(client, to_delete):
+    """Create/read/update/delete metrics."""
+    METRIC_NAME = 'topic_exists-%d' % (_millis(),)
+    DESCRIPTIION = "Robots all up in your server"
+    UPDATED_DESCRIPTION = "Danger, Will Robinson!"
+    FILTER = 'logName:apache-access AND textPayload:robot'
+
+    # [START metric_create]
+    new_metric = client.metric(METRIC_NAME,
+                               filter_=FILTER, description=DESCRIPTIION)
+    assert not new_metric.exists()  # API request
+    new_metric.create()             # API request
+    assert new_metric.exists()      # API request
```

Comment:
@tseaver I am still on team "Don't block this PR on it", but to clarify

> they aren't "testing" asserts"

I realize that, that isn't the problem. I suppose I have an `assert` allergy. But for end users my contention is that

```python
>>> new_metric.exists() 
True
```

is much more readable than `assert new_metric.exists()`.